src/plugins/presentation_util/public/plugin.ts

```diff
@@ -18,15 +18,42 @@
  */
 
 import { CoreSetup, CoreStart, Plugin } from '../../../core/public';
-import { PresentationUtilPluginSetup, PresentationUtilPluginStart } from './types';
+import { pluginServices, kibanaServiceRegistry } from './services';
+import {
+  PresentationUtilPluginSetup,
+  PresentationUtilPluginStart,
+  PresentationUtilPluginSetupDeps,
+  PresentationUtilPluginStartDeps,
+} from './types';
 
 export class PresentationUtilPlugin
-  implements Plugin<PresentationUtilPluginSetup, PresentationUtilPluginStart> {
-  public setup(core: CoreSetup): PresentationUtilPluginSetup {
-    return {};
+  implements
+    Plugin<
+      PresentationUtilPluginSetup,
+      PresentationUtilPluginStart,
+      PresentationUtilPluginSetupDeps,
+      PresentationUtilPluginStartDeps
+    > {
+  public setup(
+    coreSetup: CoreSetup<PresentationUtilPluginStart>,
+    setupPlugins: PresentationUtilPluginSetup
+  ): PresentationUtilPluginSetup {
+    const getServices = async () => {
+      if (!pluginServices.isStarted()) {
+        const [coreStart, startPlugins] = await coreSetup.getStartServices();
+        kibanaServiceRegistry.start({ coreStart, coreSetup, startPlugins, setupPlugins });
+        pluginServices.setRegistry(kibanaServiceRegistry);
+      }
+
+      return pluginServices.getServices();
+    };
+    return { getServices };
```

Comment:
If we have no reason for other plugins to need our services then we should just return an empty object here. 

But, we should proably also call `getServices()` directly here to go ahead and startup the services.

Reply:
Also, would probably be a good idea to have a conversation with someone from core to make sure this architecture won't cause any issues because I guess technically our components that we will share are stateful as they will depend on these services created by our plugin starting up and I know there can be issues with statefulness if things get split across bundles somehow.

Reply:
You were right... so I refactored in the latest iteration.